packages/react-charting/docs/TestingStrategy.md

```diff
@@ -0,0 +1,123 @@
+
+Currently, the code coverage for master branch:
+
+Commit Id:
+
+![](RackMultipart20231026-1-6nzx5m_html_e4b63f8ce1082fed.png)
+
+**Example: How to Add Tests for Donut Chart**
+
+![](RackMultipart20231026-1-6nzx5m_html_dffcd29538dede80.png)
+
+**Step 1 [Unit Tests]:**
+
+1. Identify the functions that can be unit tested (example, functions having calculations or getting values from Utils, etc).
+
+If required, extract the unit testable portions out of the functions which can be independently unit tested without any requirement of DOM elements.
+
+1. **convertToLocaleString()** [src/components/DonutChart/DonutChart.base.tsx]
+2. **getAccessibleDataObject()** [src/components/DonutChart/DonutChart.base.tsx]
+3. **\_valueInsideDonut()** [src/components/DonutChart/DonutChart.base.tsx]
+4. **wrapTextInsideDonut()** [src/components/DonutChart/Pie.tsx]
+5. **\_computeTotalValue()** [src/components/DonutChart/Pie.tsx]
+
+6. Identify the parts within those functions which cannot be tested via unit tests.
+
+7. **wrapTextInsideDonut() -** cannot be unit tested as it requires the tspan length to be calculated using Browser Functions like **getComputedTextLength()**
+8. **\_computeTotalValue() -** depends on the data prop passed down from the DonutChart.base to Pie during component rendering
+
+**Step 2 [Component Tests]:**
+
+1. Identify the component level scenarios that can be tested (as per the test plan). - Component Test Plan
+2. Identify if the parts in Step 1 (ii) can be covered via component tests.
+
+3. **\_computeTotalValue() -** can be covered by component tests by passing the data prop during component rendering.
+
+4. Identify the scenarios which cannot be covered via Step 2(I and ii).
+
+5. **wrapTextInsideDonut() -** cannot be unit/component tested as it requires the tspan length to be calculated using Browser Functions like **getComputedTextLength()**
+
+**Step 3 [E2E Tests]:**
+
+1. Identify the E2E scenarios that can be covered (as per the test plan) - Visual Regression with Interaction Test Plan – Donut Chart
+2. Cover all the test scenarios which could not be covered either in Step 1 or in Step 2.
+
+3. **wrapTextInsideDonut() -** can be covered by E2E tests as the component is now rendered in a browser.
+
+Sample PRs and Test Plans:
+
+1. Unit test plan and unit tests: [Adding unit tests for donut chart by srmukher · Pull Request #27424 · microsoft/fluentui (github.com)](https://github.com/microsoft/fluentui/pull/27424/files)
+
+Test Plan: [https://microsoftapc-my.sharepoint.com/:w:/g/personal/srmukher_microsoft_com/EZffGTRDJDBBg3nb1CnF8koB3RDDYHxWErXdp7StPqKZeQ?e=72HMsD](https://microsoftapc-my.sharepoint.com/:w:/g/personal/srmukher_microsoft_com/EZffGTRDJDBBg3nb1CnF8koB3RDDYHxWErXdp7StPqKZeQ?e=72HMsD)
+
+1. Component test plan and component tests: [Component testing - Donut chart by srmukher · Pull Request #27033 · microsoft/fluentui (github.com)](https://github.com/microsoft/fluentui/pull/27033)
+
```

Comment:
I believe these test plans have already been added to the docs section in our repo.
Update internal links with public links.

Reply:
updated